pkg/cloud/aws/actuators/machine/actuator.go

```diff
@@ -289,6 +305,66 @@ func (a *Actuator) Delete(ctx context.Context, cluster *clusterv1.Cluster, machi
 	return nil
 }
 
+func (a *Actuator) cordon(ctx context.Context, cluster *clusterv1.Cluster, machine *clusterv1.Machine, scope *actuators.MachineScope) (error, error) {
```

Comment:
Returning two error values seems a bit odd to me, why not just a single error?

Reply:
This is because I'm using these method that returns two erros. To be honest it was odd to me too:

https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/840be1a5ffda479425d2a4cf61d9b81be47f389c/vendor/k8s.io/kubernetes/pkg/kubectl/drain/cordon.go#L75:24

Reply:
The 2nd error is non-nil if it was unable to create the two-way merge patch.

Honestly, the `CordonHelper` is small enough that I would probably recommend implementing its logic here. Also, I'm not sure we should be letting the helper send an "update" request for the Node in the event we got an error trying to create the merge patch. Patch >>> Update.